torch/csrc/Module.cpp

```diff
@@ -346,11 +346,49 @@ PyObject* THPModule_swap_tensor_impl(PyObject* _unused, PyObject* args) {
   THPVariable* a = reinterpret_cast<THPVariable*>(a_);
   THPVariable* b = reinterpret_cast<THPVariable*>(b_);
 
+  TORCH_CHECK(
+      a->cdata->use_count() == 1,
+      "Expected single reference to a's Tensor object but got ",
+      a->cdata->use_count());
+  TORCH_CHECK(
+      b->cdata->use_count() == 1,
+      "Expected single reference to b's Tensor object but got ",
+      b->cdata->use_count());
+  // weak_use_count() adds 1 if use_count is non-zero
+  TORCH_CHECK(
```

Comment:
There is  a weak_ref check  on the Pyobj, do `a->cdata->weak_use_count()` and `weakref.getweakrefs(t)` differ?

I am thinking the answer is yes because `a->cdata->weak_use_count()` probably counts cpp and python weakrefs while I think `weakref.getweakrefs(t)` would only count the python ones.

Is there a way to install a cpp-only weakref to exercise this check?